### PR TITLE
Add matrix of Xcode/Swift versions to CI

### PR DESCRIPTION
**Pull request by erichoracek**
This enables us to check that Lottie compiles in Swift 5.3, 5.4, and 5.5, which has regressed with recent changes on `master`.

We also add a local setup action to reduce the setup code that's interspersed through all of the steps in the same style as the one added to Epoxy here https://github.com/airbnb/epoxy-ios/pull/74

Additionally fixes those Swift 5.3/5.4 compilation errors that prevent the project from building on those Xcode versions.

### CODE REVIEW

File: Example/Example.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -507,6 +508,7 @@
 				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = YES;
 				CLANG_WARN_UNGUARDED_AVAILABILITY = YES_AGGRESSIVE;
 				CODE_SIGN_ENTITLEMENTS = macOS/macOS.entitlements;
+				CODE_SIGN_IDENTITY = "-";
```
Comment (erichoracek) [inline Example/Example.xcodeproj/project.pbxproj:511]: This change was added by Xcode 13.2

File: Example/Example.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -665,6 +668,7 @@
 				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
 				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
+				CLANG_WARN_QUOTED_INCLUDE_IN_FRAMEWORK_HEADER = YES;
```
Comment (erichoracek) [inline Example/Example.xcodeproj/project.pbxproj:671]: This change was added by Xcode 13.2

File: Example/iOS/ViewControllers/SampleListViewController.swift [hunk 3]
```diff
@@ -112,7 +110,7 @@ final class SampleListViewController: CollectionViewController {
   private func configureSettingsMenu() {
     navigationItem.rightBarButtonItem = UIBarButtonItem(
       title: "Settings",
-      image: .init(systemName: "gear"),
+      image: UIImage(systemName: "gear")!,
```
Comment (erichoracek) [inline Example/iOS/ViewControllers/SampleListViewController.swift:113]: Seems like we need to force unwrap here, not sure how this is compiling as the method takes a non-optional image

Comment (calda) [reply Example/iOS/ViewControllers/SampleListViewController.swift:113]: weird!

File: Sources/Private/Experimental/Layers/ImageLayer.swift [hunk 4]
```diff
@@ -24,12 +24,12 @@ final class ImageLayer: BaseCompositionLayer {
   /// Called by CoreAnimation to create a shadow copy of this layer
   /// More details: https://developer.apple.com/documentation/quartzcore/calayer/1410842-init
   override init(layer: Any) {
-    guard let layer = layer as? Self else {
```
Comment (calda) [inline Sources/Private/Experimental/Layers/ImageLayer.swift:27]: Interesting, didn't know this used to be unsupported

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -10,58 +10,100 @@ jobs:
   build-package:
     name: "Build Package"
     runs-on: macos-latest
+    strategy:
+      matrix:
+        xcode:
+        - '12.4' # Swift 5.3
```
Comment (calda) [inline .github/workflows/main.yml:16]: Why do we need to support Swift 5.3? Just curious

**Review by calda — APPROVED** [review]

**Review by calda — APPROVED** [review]
Thanks!

File: .github/workflows/main.yml (hunk 5 above)
Comment (erichoracek) [reply .github/workflows/main.yml:16]: We don't have to, it was just previously supported in `master` based on https://swiftpackageindex.com/airbnb/lottie-ios so I continued to support it

Comment (calda) [reply .github/workflows/main.yml:16]: Good to know!

**Comment by calda** [comment]
Maybe we should update `README.md`, where it currently says:

> Lottie is written in Swift 5.4.

To something like:

> Lottie supports Swift 5.3 and later.

File: Sources/Private/Experimental/Layers/ImageLayer.swift (hunk 4 above)
Comment (erichoracek) [reply Sources/Private/Experimental/Layers/ImageLayer.swift:27]: This was supported, it's the fact that we're using `layer` in the fatalError` that was unsupported

Comment (calda) [reply Sources/Private/Experimental/Layers/ImageLayer.swift:27]: oooh I see

**Comment by erichoracek** [comment]
> Maybe we should update `README.md`, where it currently says:
> 
> > Lottie is written in Swift 5.4.
> 
> To something like:
> 
> > Lottie supports Swift 5.3 and later.

Good point, I added a Swift Package Index badge here 9a7b7e086292fdc1b7a1e4a5c9ed45054b7f6352

**Pull request merged**